internal/applier/helmer.go

```diff
@@ -283,3 +201,98 @@ func createClientWithToken(cfg *rest.Config, token string) (*kubernetes.Clientse
 
 	return clientSet, nil
 }
+
+// Looks for the ConfigSources by name in the install namespace and gathers values
+func processConfig(ctx context.Context, tokenGetter *authentication.TokenGetter, ext *ocv1alpha1.ClusterExtension, values chartutil.Values) (chartutil.Values, error) {
```

Comment:
This function is generally pretty straightforward but I wonder if the mental model here could be simplified a bit with some logical abstractions or separate functions?

I think the discriminated union approach for the API might also make that abstraction/functional separation a bit nicer.

Maybe something like:
```go 
func processConfigSources(ctx context.Context, client *kubernetes.Clientset, configSources []ocv1alpha1.ConfigSource, values chartutil.Values) (chartutil.Values, error) {
  for _, configSource := range configSources {
    userSuppliedValues, err := loadValuesFromConfigSource(ctx, client, configSource)
    // err check
    // do something with values
  }
  // return loaded values
}

func loadValuesFromConfigSource(ctx context.Context, client *kubernetes.Clientset, configSource ocv1alpha1.ConfigSource) (chartutil.Values, error) {
  switch configSource.Type {
    case ocv1alpha1.ConfigSourceTypeConfigMap:
      // Load from configmap
    case ocv1alpha1.ConfigSourceTypeSecret:
      // Load from secret
    case ocv1alpha1.ConfigSourceTypeInline:
      // Load from inline
  }
}
```

What I put might not work as is, but hopefully gets the point across that I think some responsibility separation here might be useful for readability, maintenance, and testing.